src/extension.ts

```diff
@@ -118,7 +120,13 @@ export class Extension implements RunHooks {
     });
     this._testController = vscode.tests.createTestController('playwright', 'Playwright');
     this._testController.resolveHandler = item => this._resolveChildren(item);
-    this._testController.refreshHandler = () => this._rebuildModels(true).then(() => {});
+    this._testController.refreshHandler = async () => {
+      // all other rebuilds go through the batch, this one does not.
+      // refreshHandler is only called on user gesture,
+      // and it's a "last resort" reload that's only used when something is broken.
+      // We want it to run immediately, even if it's prone to race conditions.
```

Comment:
Can we set some flag so that all other `_rebuildModels()` will exit early, and avoid the race? Perhaps a cancellation token would be handy for this.